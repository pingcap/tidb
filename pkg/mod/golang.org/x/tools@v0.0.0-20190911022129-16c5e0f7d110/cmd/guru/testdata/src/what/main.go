package main // @what pkgdecl "main"

// Tests of 'what' queries.
// See go.tools/guru/guru_test.go for explanation.
// See what.golden for expected query results.

func main() {
	f()             // @what call "f"
	var ch chan int // @what var "var"
	<-ch            // @what recv "ch"
}
