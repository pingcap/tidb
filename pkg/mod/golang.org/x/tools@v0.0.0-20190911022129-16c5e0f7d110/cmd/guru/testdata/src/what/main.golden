-------- @what pkgdecl --------
identifier
source file
modes: [definition describe freevars implements pointsto referrers whicherrs]
srcdir: testdata/src
import path: what

-------- @what call --------
identifier
function call
expression statement
block
function declaration
source file
modes: [callees callers callstack definition describe freevars implements pointsto referrers whicherrs]
srcdir: testdata/src
import path: what

-------- @what var --------
variable declaration
variable declaration statement
block
function declaration
source file
modes: [callers callstack describe freevars pointsto whicherrs]
srcdir: testdata/src
import path: what

-------- @what recv --------
identifier
unary <- operation
expression statement
block
function declaration
source file
modes: [callers callstack definition describe freevars implements peers pointsto referrers whicherrs]
srcdir: testdata/src
import path: what
ch
ch

